### ISSUE TITLE
LTO warnings

### ISSUE THREAD

**Issue by kloczek** [libical/libical#363]
During linking with LTo optimisation I see two warnings:
```
cd /home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss && /usr/bin/cmake -E cmake_link_script CMakeFiles/icalss.dir/link.txt --verbose=1
/usr/bin/gcc -fPIC -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switch
es -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection
 -fcf-protection -flto -fvisibility=hidden -Wno-deprecated -Wall -Wno-unknown-pragmas -Wextra -Winit-self -Wmissing-include-dirs -Wunused -Wno-div-by-zero -Wundef -Wpoin
ter-arith -Wtype-limits -Wwrite-strings -Werror=return-type -Wunused-but-set-variable -Wlogical-op -Wsizeof-pointer-memaccess -Wformat -Wformat-security -Wredundant-decl
s -D_XOPEN_SOURCE=500 -D_DEFAULT_SOURCE -D_GNU_SOURCE -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto -fuse-linker-plugin -s
hared -Wl,-soname,libicalss.so.3 -o ../../lib/libicalss.so.3.0.3 CMakeFiles/icalss.dir/icalcalendar.c.o CMakeFiles/icalss.dir/icalclassify.c.o CMakeFiles/icalss.dir/ical
cluster.c.o CMakeFiles/icalss.dir/icalgauge.c.o CMakeFiles/icalss.dir/icaldirset.c.o CMakeFiles/icalss.dir/icalfileset.c.o CMakeFiles/icalss.dir/icalset.c.o CMakeFiles/i
calss.dir/icalspanlist.c.o CMakeFiles/icalss.dir/icalmessage.c.o CMakeFiles/icalss.dir/icalsslexer.c.o CMakeFiles/icalss.dir/icalssyacc.c.o CMakeFiles/icalss.dir/icalbdb
set.c.o -Wl,-rpath,/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/lib: ../../lib/libical.so.3.0.3 -ldb -lpthread -licuuc -licui18n
/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss/icalssyacc.c:173:5: warning: type of 'sslex' does not match original declaration [-Wlto-type-mismatch]
 int sslex(void *YYPARSE_PARAM);
     ^
/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss/icalsslexer.c:764:1: note: type mismatch in parameter 1
 YY_DECL
 ^
/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss/icalsslexer.c:764:1: note: 'sslex' was previously declared here
/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss/icalgauge.c:34:12: warning: type of 'ssparse' does not match original declaration [-Wlto-type-mismatch]
 extern int ssparse(void);
            ^
/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss/icalssyacc.c:1166:1: note: type mismatch in parameter 1
 yyparse (void *YYPARSE_PARAM)
 ^
/home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss/icalssyacc.c:1166:1: note: 'ssparse' was previously declared here
cd /home/tkloczko/rpmbuild/BUILD/libical-3.0.3/src/libicalss && /usr/bin/cmake -E cmake_symlink_library ../../lib/libicalss.so.3.0.3 ../../lib/libicalss.so.3 ../../lib/l
ibicalss.so

```

**Comment by winterz** [comment]
please tell me which CC flags you added to the build

**Comment by kloczek** [comment]
```
CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto'

LDFLAGS='-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto -fuse-linker-plugin'
````
However it doesn't matter because LTO linking usually when emits [-Wlto-type-mismatch] warnings it exposes some exact types of differences between declarations and places where functions are defined.
In this case in one place is used int and in other YY_DECL.

**Comment by winterz** [comment]
fixed in commit c2c99e7eb54904d91b909e2053c0a896c276471b
